fetchLater corrections and updates (#39118)
mdn/content: files/en-us/web/api/fetchlater_api/fetchlater_quotas/index.md

2. `<iframe src="https://b.com/">` receives 8KiB of the default shared quota.
3. The 8KiB is not transferred to `c.com` when `<iframe src="https://b.com/">` redirects to there, but the 8KiB is not released.

### Sandboxed same-origin iframes are effectively separate origins

> Given this iframe embedded on `http://www.example.com`:

Reviewer: I would prefer using HTTPS unless it's intended

> Given this iframe embedded on `https://www.example.com`:

Reviewer 2: ```suggestion
As an example, the following `<iframe>` is embedded on `https://www.example.com`:
```

Author: Seems rather final. Let's add an "if" to your suggestion:

```suggestion
As an example, if the following `<iframe>` is embedded on `https://www.example.com`:
```

```html
<iframe src="https://www.example.com/iframe" sandbox="allow-scripts"></iframe>
```

> This would not be seem as "same-origin" despite being hosted on the same origin as the top-level dcument as it is in it's own sandbox iframe. Therefore, by default, it should be allocated 8KiB quota from the shared 128KiB quota.

Reviewer 2: ```suggestion
This would not be considered "same-origin", despite being hosted on the same origin, as the top-level document is in its own sandboxed `<iframe>`. Therefore, by default, it should be allocated an 8KiB quota from the total shared 128KiB quota.
```

Reviewer 2: Also, a question — how do we know the top-level document is in its own sandboxed `<iframe>` (if that is indeed what you meant, which I think it is)? There is nothing that states this in the example, unless I am missing something.

Author: Yeah think you've misunderstood here (which shows it wasn't clear enough!). Tweaking your suggestion:

```suggestion
This would not be considered "same-origin", despite being hosted on the same origin as the top-level document, as the `<iframe>` is in a sandboxed environment. Therefore, by default, it should be allocated an 8KiB quota from the total shared 128KiB quota.
```

### Disallowing `fetchLater()` from iframes

> You can use the `allow` attribute of `<iframe>`s to prevent `fetchLater()` quota from being allocated to the iframe:

Reviewer: ```suggestion
You can use the `allow` attribute of `<iframe>` elements to prevent `fetchLater()` quota from being allocated to the iframe:
```

> You can use the `allow` attribute of `<iframe>` elements to prevent `fetchLater()` quota from being allocated to the iframe:

Reviewer 2: ```suggestion
You can use the `<iframe>` [`allow`](/en-US/docs/Web/HTML/Reference/Elements/iframe#allow) attribute to prevent `fetchLater()` quota from being allocated to the `<iframe>`:
```

```html
<iframe
  src="https://www.example.com/iframe"
  allow="deferred-fetch;deferred-fetch-minimal;"></iframe>
```

> Where `allow="deferred-fetch"` is needed to prevent same-origin iframes using up the 512KiB quota, and `allow="deferred-fetch-minimal"` is needed to prevent cross-origin iframes using up the 128KiB quota, and including both will prevent either being used no matter what the `src`.

Reviewer 2: ```suggestion
The `allow="deferred-fetch"` directive is needed to prevent same-origin `<iframe>`s from using up the 512KiB quota, and the `allow="deferred-fetch-minimal"` directive is needed to prevent cross-origin `<iframe>`s from using up the 128KiB quota. Including both will prevent both from being used, regardless of the `src` value.
```

> The `allow="deferred-fetch"` directive is needed to prevent same-origin iframes from using up the 512KiB quota, and the `allow="deferred-fetch-minimal"` directive is needed to prevent cross-origin iframes from using up the 128KiB quota. Including both will prevent both from being used, regardless of the `src` value.

Reviewer 2: ```suggestion
The `allow="deferred-fetch"` directive is needed to prevent same-origin iframes from using up the 512KiB quota, and the `allow="deferred-fetch-minimal"` directive is needed to prevent cross-origin iframes from using up the 128KiB quota. Including both directives will prevent both quotas from being used, regardless of the `src` value.
```

fetchLater("https://a.example.com", { method: "POST", body: a_40kb_body });
```

### Redirects of subframes back to the top-level origin allow use of the top-level quota

> Assuming a top-level document on `a.com`, which embeds a `<iframe src="https://b.com/">`, which redirects to `a.com`, and no explicit top-level Permission Policies.

Reviewer 2: ```suggestion
Assuming a top-level document at `a.com`, which embeds `<iframe src="https://b.com/">`, which redirects to `a.com`, and no explicit top-level Permission Policies:
```

1. The top-level frame of `a.com` has the default 512KiB quota.

> 2. `<iframe src="https://b.com/">` receives 8KiB of the default shared quota.

Reviewer 2: What is the total available shared quota again? Might be a good idea to mention this.

Author: ```suggestion
2. `<iframe src="https://b.com/">` receives 8KiB of the default shared quota of 128KiB.
```

> 3. The 8KiB is not transferred to `a.com` when `<iframe src="https://b.com/">` redirects to there, but it is able to share the full top-level quota again, and the 8KiB is released.

Reviewer 2: ```suggestion
3. The 8KiB is not transferred to `a.com` when `<iframe src="https://b.com/">` redirects to there, but it is able to share the full top-level quota again, and the previously-shared 8KiB quota is released.
```

Author: Let's go with "previously-allocated"

```suggestion
3. The 8KiB is not transferred to `a.com` when `<iframe src="https://b.com/">` redirects to there, but it is able to share the full top-level quota again, and the previously-allocated 8KiB quota is released.
```
